mavros_extras/src/plugins/landing_target.cpp

```diff
@@ -320,67 +396,77 @@ class LandingTargetPlugin : public plugin::PluginBase,
 
     auto rpy = ftf::quaternion_to_rpy(orientation);
 
-    ROS_DEBUG_THROTTLE_NAMED(
-      10, "landing_target", "Rx landing target: "
-      "ID: %d frame: %s angular offset: X:%1.3frad, Y:%1.3frad) "
-      "distance: %1.3fm position: X:%1.3fm, Y:%1.3fm, Z:%1.3fm) "
-      "orientation: roll:%1.4frad pitch:%1.4frad yaw:%1.4frad "
-      "size: X:%1.3frad by Y:%1.3frad type: %s",
-      land_target.target_num, utils::to_string(static_cast<MAV_FRAME>(land_target.frame)).c_str(),
-      land_target.angle_x, land_target.angle_y, land_target.distance,
-      position.x(), position.y(), position.z(), rpy.x(), rpy.y(),
-      rpy.z(), land_target.size_x, land_target.size_y,
-      utils::to_string(static_cast<LANDING_TARGET_TYPE>(land_target.type)).c_str());
-
-    auto pose = boost::make_shared<geometry_msgs::PoseStamped>();
-    pose->header = m_uas->synchronized_header(frame_id, land_target.time_usec);
-
-    tf::pointEigenToMsg(position, pose->pose.position);
-    tf::quaternionEigenToMsg(orientation, pose->pose.orientation);
-
-    land_target_pub.publish(pose);
-
-    if (send_tf) {
-      geometry_msgs::TransformStamped transform;
-
-      transform.header.stamp = pose->header.stamp;
-      transform.header.frame_id = "landing_target_" + boost::lexical_cast<std::string>(
+    RCLCPP_DEBUG_STREAM_THROTTLE(
+      uas->get_logger(),
+      *uas->get_clock(), 10,
+      "landing_target:\n" <<
+      "\tRx landing target:" <<
+      "\n\t\tID: " << land_target.target_num <<
+      "\n\t\tframe: " << utils::to_string(static_cast<MAV_FRAME>(land_target.frame)).c_str() <<
+      "\n\t\tangular offset: X: " << land_target.angle_x << "rad, Y: " << land_target.angle_y << "rad" <<
+      "\n\t\tdistance: " << land_target.distance << "m" <<
+      "\n\t\tposition:" <<
+      "\n\t\t\tX: " << position.x() << "m" <<
+      "\n\t\t\tY: " << position.y() << "m" <<
+      "\n\t\t\tZ: " << position.z() << "m" <<
+      "\n\t\torientation:" <<
+      "\n\t\t\troll:  " << rpy.x() << "rad" <<
+      "\n\t\t\tpitch: " << rpy.y() << "rad" <<
+      "\n\t\t\tyaw:   " << rpy.z() << "rad" <<
+      "\n\t\tsize: X: " << land_target.size_x << "rad, Y: " << land_target.size_y << "rad" <<
+      "\n\t\ttype: " << utils::to_string(static_cast<LANDING_TARGET_TYPE>(land_target.type)).c_str() << std::endl);
```

Comment:
NOTE: c++11 messages have `to_yaml()` method.

Reply:
That's great to know! Check newest commit :) I removed the debugging before we send out the MAVLink LANDING_TARGET message. In mavros/ros2 all MAVLink messages that we try to send get published to /mavlink/to, right? So if someone is interested in debugging a particular message, it's easy to write a small script to attach to /mavlink/to, which makes it unnecessary to do MAVLink debugging within the plugins themselves. Or what do you think?

Reply:
With debugging there are two notes:
1. In ROS2 you can have multiple UAS links, and yes, all plugins now pushing to appropriate topic. But names are different, to/from terms changed to source/sink.
2. In plugin you usually have decoded message, which is much more human readable with to_yaml(), but with a script you could inspect whole traffic of the UAS path (FCU<-(Router)->UAS).

https://github.com/mavlink/mavros/blob/cf15f80f9ee6d1b5ab0ffd665207bc1947e4ed41/mavros/src/lib/mavros_router.cpp#L476-L481

Reply:
So ROS1 `/mavlink/to` maps to `/uas1/mavlink_sink` (when uas_url is `/uas1`).

Reply:
Also note that:

https://github.com/mavlink/mavros/blob/ros2/mavros/src/lib/mavros_router.cpp#L471